share/man/man9/cdefs.9

```diff
@@ -0,0 +1,351 @@
+.It Sy __trylocks_shared Ta
+.It Sy __unlocks Ta
+.It Sy __asserts_exclusive Ta
+.It Sy __asserts_shared Ta
+.It Sy __requires_exclusive Ta
+.It Sy __requires_shared Ta
+.It Sy __requires_unlocked Ta
+.It Sy __no_lock_analysis Ta
+.It Sy __nosanitizeaddress Ta
+.It Sy __nosanitizememory Ta
+.It Sy __nosanitizethread Ta
+.It Sy __nostackprotector Ta
+.It Sy __guarded_by(x) Ta
+.It Sy __pt_guarded_by(x) Ta
+.El
+.Ss Emulated Keywords
+As C evolves, many of the old macros we once used have been incorporated into
+the standard language.
+As this happens, we add support for these keywords as macros for older
+compilation environments.
+Sometimes this results in a nop in the older environment.
+.Bl -column "---------------"
+.It Sy Keyword Ta Sy Description
+.It Sy _Alignas(x) Ta
+.It Sy _Alignof(x) Ta
+.It Sy _Noreturn Ta Expandsd to
+.Dq [[noreturn]]
+in C++-11 and newer compilation environments, otherwise
+.Dq __dead2
+.It Sy _Static_assert(x, y) Ta Compile time assertion that
+.Fa x
+is true, otherwise emit
+.Fa y
+as the error message.
+.It Sy _Thread_local Ta Designate variable as thread local storage
+.It Sy __generic Ta implement _Generic-like features which aren't entirely possible to emulate the _Generic keyword
+.It Sy __noexcept Ta to emulate the C++11 argument-less noexcept form
+.It Sy __noexcept_if Ta to emulate the C++11 conditional noexcept form
+.It Sy _Nonnull Ta
+.It Sy _Nullable Ta
+.It Sy _Null_unspecified Ta
+.El
+.Ss Support Macros
+The following macros are defined, or have specific values, to denote certain
+things about the build environment.
+.Bl -column "---------------"
+.It Sy Macro Ta Sy Description
+.It Sy __LONG_LONG_SUPPORTED Ta Variables may be declared
+.Dq long long .
+This is defined for C99 or newer and C++ environments.
+.It Sy __STDC_LIMIT_MACROS Ta
+.It Sy __STDC_CONSTANT_MACROS Ta
+.El
+.Ss Nice to have macros
+These macros make it easier to do a number of things, even though strictly
+speaking the standard places their normal form in another header.
+.Bl -column "---------------"
+.It Sy Macro Ta Sy Description
+.It Sy __offsetof(type,field) Ta
+.It Sy __rangeof(type,start,end) Ta
+.It Sy __containerof(x,s,m) Ta
+.El
+.Ss ID Strings
+This section is deprecated, but is kept around because too much contrib software
+still uses these.
+.Bl -column "---------------"
+.It Sy Macro Ta Sy Description
+.It Sy __IDSTRING(name,string) Ta
+.It Sy __FBSDID(s) Ta
+.It Sy __RCSID(s) Ta
+.It Sy __RCSID_SOURCE(s) Ta
+.It Sy __SCCSID(s) Ta
+.It Sy __COPYRIGHT(s) Ta
+.El
+.Sh Supported C Environments
+.Fx
+supports a number C standard environments.
+Selection of the language dialect is a compiler-dependent command line option,
+though it is usually
+.Fl cstd=XX
+where XX is the standard to set for compiling, such as c89 or c23.
+.Fx
+provides a number of selection macros to control visibility of symbols.
+Please see the section on Selection Macros for the specifics.
+.Pp
+.Bl -tag -width two-column -compact
```

Comment:
```suggestion
.Bl -tag
```
I think this is the superior formatting, for this table in particular (dropping `-compact`). The spacing makes sense for these multi-paragraph blocks:
```
Supported C Environments
     FreeBSD supports a number C standard environments.  Selection of the
     language dialect is a compiler-dependent command line option, though it
     is usually -cstd=XX where XX is the standard to set for compiling, such
     as c89 or c23.  FreeBSD provides a number of selection macros to control
     visibility of symbols.  Please see the section on Selection Macros for
     the specifics.

     K & R   Pre-ANSI Kernighan and Ritchie C.  Sometimes called “knr” or
             “C78” to distinguish it from newer standards.  Support for this
             compilation environment is dependent on compilers supporting this
             configuration.  Most of the old forms of C have been deprecated
             or removed in ISO/IEC 9899:2024 (“ISO C23”).  Compilers make
             compiling in this mode increasingly difficult and support for it
             may ultimately be removed from the tree.

     ANSI X3.159-1989 (“ANSI C89”)
             __STDC__ is defined, however __STDC_VERSION__ is not.

             Strict environment selected with _ANSI_SOURCE.

     ISO/IEC 9899:1999 (“ISO C99”)
             __STDC_VERSION__ = 199901L
...
```